build.gradle

```diff
@@ -147,18 +155,26 @@ subprojects {
 
     publishing {
         publications {
-            mavenJava(MavenPublication) {
+            shadow(MavenPublication) {
                 from components.java
 
-                // These are for unpublished test classes
-                suppressPomMetadataWarningsFor('testFixturesApiElements')
-                suppressPomMetadataWarningsFor('testFixturesRuntimeElements')
+                project.shadow.component(it)
 
                 // Customize artifact ID for core project
                 if (project.name == 'opensearch-remote-metadata-sdk-core') {
                     artifactId = 'opensearch-remote-metadata-sdk'
                 }
 
+                artifacts.clear()
```

Comment:
Needed to remove sources and javadoc jar artifacts that may have been built already before the shadow publication, otherwise they'd cause duplicate errors with the lines below.